<h1>Car Price Prediction</h1>


In [15]:
# Author Pablo Morales

In [16]:
#This is one of my practice of Data Analysis at Coursera.

<h5>Questions</h5> 

Do I know if the dealer is offering fair value for my trade-in?
Do I know if I put a fair value on my car?

<h3>Model Development</h3>


I will develop several models that will predict the price of the car using the variables.

<h3>Import Libraries</h3>

In [1]:
import pandas as pd  
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt  

In [3]:
#This dataset was hosted on IBM Cloud object.

In [4]:
path = 'https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DA0101EN-SkillsNetwork/labs/Data%20files/automobileEDA.csv'

In [5]:

df = pd.read_csv(path)
df.head(10)

symboling  normalized-losses         make aspiration num-of-doors  \
0          3                122  alfa-romero        std          two   
1          3                122  alfa-romero        std          two   
2          1                122  alfa-romero        std          two   
3          2                164         audi        std         four   
4          2                164         audi        std         four   
5          2                122         audi        std          two   
6          1                158         audi        std         four   
7          1                122         audi        std         four   
8          1                158         audi      turbo         four   
9          2                192          bmw        std          two   

    body-style drive-wheels engine-location  wheel-base    length  ...  \
0  convertible          rwd           front        88.6  0.811148  ...   
1  convertible          rwd           front        88.6  0.811148  ...   
2    hatchback          rwd           front        94.5  0.822681  ...   
3        sedan          fwd           front        99.8  0.848630  ...   
4        sedan          4wd           front        99.4  0.848630  ...   
5        sedan          fwd           front        99.8  0.851994  ...   
6        sedan          fwd           front       105.8  0.925997  ...   
7        wagon          fwd           front       105.8  0.925997  ...   
8        sedan          fwd           front       105.8  0.925997  ...   
9        sedan          rwd           front       101.2  0.849592  ...   

   compression-ratio  horsepower  peak-rpm city-mpg highway-mpg    price  \
0                9.0       111.0    5000.0       21          27  13495.0   
1                9.0       111.0    5000.0       21          27  16500.0   
2                9.0       154.0    5000.0       19          26  16500.0   
3               10.0       102.0    5500.0       24          30  13950.0   
4                8.0       115.0    5500.0       18          22  17450.0   
5                8.5       110.0    5500.0       19          25  15250.0   
6                8.5       110.0    5500.0       19          25  17710.0   
7                8.5       110.0    5500.0       19          25  18920.0   
8                8.3       140.0    5500.0       17          20  23875.0   
9                8.8       101.0    5800.0       23          29  16430.0   

  city-L/100km  horsepower-binned  diesel  gas  
0    11.190476             Medium       0    1  
1    11.190476             Medium       0    1  
2    12.368421             Medium       0    1  
3     9.791667             Medium       0    1  
4    13.055556             Medium       0    1  
5    12.368421             Medium       0    1  
6    12.368421             Medium       0    1  
7    12.368421             Medium       0    1  
8    13.823529             Medium       0    1  
9    10.217391                Low       0    1  

[10 rows x 29 columns]

<h3>Linear Regression and Multiple Linear Regression</h3>

In [6]:
from sklearn.linear_model import LinearRegression

In [7]:
#Create the linear regression object:

In [8]:
lm = LinearRegression()
lm

LinearRegression()

In [10]:
X = df[['highway-mpg']]
Y = df['price']

<h3>How could "highway-mpg" help us predict car price?</h3>

In [11]:
lm.fit(X,Y)

LinearRegression()

In [12]:
Yhat=lm.predict(X)
Yhat[0:5] 

array([16236.50464347, 16236.50464347, 17058.23802179, 13771.3045085 ,
       20345.17153508])

<h3>What is the value of the intercept (a)?</h3>

In [13]:
lm.intercept_ 

38423.3058581574

<h3>What is the value of the slope (b)?</h3>

In [14]:
lm.coef_

array([-821.73337832])

<h3>What is the final estimated linear model we get?</h3>

<h5>Price = 38423.31 - 821.73 x highway-mpg</h5>